pkg/broker/binding_operations.go

```diff
@@ -141,36 +139,31 @@ func (b Broker) userFromParams(bindingID string, password string, serviceID stri
 		}
 	}
 
-	// If the plan ID is specified we construct the provider object from the service and plan.
-	// The plan ID is required for binding
-	if planID != "" {
-		provider, err := b.findProviderByServiceID(serviceID)
-		if err != nil {
-			return nil, err
-		}
+	// The service_id and plan_id are required to not be empty and have an invalid format.
```

Comment:
How about `// The service ID and plan ID are required to be valid per the specification, despite not being used for bindings. We look them up to ensure they can be found in the catalog.`. Just to be clear what valid means in the context of the spec. I don't think we need to mention that they must be non-empty as the `brokerapi` takes care of that.